samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -139,14 +139,21 @@ void BLEConnectivityManager::ConnectionHandler(bt_conn *conn, uint8_t conn_err)
 				Instance().Reconnect(reinterpret_cast<BLEBridgedDeviceProvider *>(context));
 			},
 			reinterpret_cast<intptr_t>(providerToRecover));
+		/* We have still a device to recover, keep the LostDevice state active */
+		Instance().UpdateStateFlag(State::LostDevice, true);
 	} else if (!sys_slist_is_empty(&Instance().mRecovery.mListToRecover)) {
 		/* There are pending providers to recover and no more scanned ones, schedule next scan operation. */
 		Instance().mRecovery.StartTimer();
+	} else if (provider->IsInitiallyConnected()) {
```

Comment:
Why do we need `provider->IsInitiallyConnected()` check here? Wouldn't `else` be enough?

Reply:
It isn't enough because without that we are going to clear the `LostDevice` state on each connection. So when we lose a device, the state is active, and then we connect another one, the state will be cleared, but we still haven't recovered the previous one.

Reply:
But why? Above we have two checks that make sure the list of devices to reconnect and the list of devices to recover are both empty. The else condition simply means there is nothing to be recovered, so why don't have to care if this device was previously connected or it is connected for the first time.